### PR TITLE
Textfield icons

### PR DESCRIPTION
**Pull request by garrett-thompson**
Implemented both trailing & leading icons in textfields. Use `trailingIcon` or `leadingIcon` prop, passing the name of the material icon

### CODE REVIEW

**Comment by mellogarrett** [comment]
Fixes #370

**Review by prateekbh — CHANGES_REQUESTED** [review]
Can we use `<Icon>` instead?

**Comment by mellogarrett** [comment]
The material-components-web docs use the `<i>` tag, but I can definitely change it if you think we should.

https://material.io/components/web/catalog/input-controls/text-fields/#leading-and-trailing-icons

**Comment by prateekbh** [comment]
@mellogarrett doesnt `<Icon>` renders to `<i>`?

**Comment by mellogarrett** [comment]
Oh sorry, I misunderstood. I'll make that change.

**Comment by prateekbh** [comment]
last thing, can you bump the version?

File: package.json [hunk 1]
```diff
@@ -90,14 +90,8 @@
...
-    "index.js": [
-      "prettier --write",
-      "git add"
-    ],
-    "**/*.jsx": [
-      "prettier --write",
-      "git add"
-    ]
+    "index.js": ["prettier --write", "git add"],
+    "**/*.jsx": ["prettier --write", "git add"]
```
Comment (filoozom) [inline package.json]: Was this changed by `npm run prettier:write`?

**Review by prateekbh — APPROVED** [review]

File: TextField/TextField.jsx [hunk 2]
```diff
@@ -75,6 +76,19 @@ class TextFieldInput extends MaterialComponent {
   materialDom(allprops) {
     let { className, ...props } = allprops;
```
Comment (prateekbh) [inline TextField/TextField.jsx]: @mellogarrett can we just do `className = className || ''`.
This will clean up nested conditions

Comment (mellogarrett) [reply TextField/TextField.jsx]: I don't know how I would write that in this case... `className = className + "..." || "..."` doesn't work

Comment (prateekbh) [reply TextField/TextField.jsx]: just `className = className || ''`, below line 76 doesn't work?
className should be a simple string isnt it?

Comment (mellogarrett) [reply TextField/TextField.jsx]: It is a string, but if it exists we want to add to it. If it's null or undefined we want to set it. Right?

Comment (prateekbh) [reply TextField/TextField.jsx]: so the code above will just initialize an empty string if its null or will leave it as is if its there

Comment (mellogarrett) [reply TextField/TextField.jsx]: Oh, duh... sorry, I get it now.

Comment (mellogarrett) [reply TextField/TextField.jsx]: I was thinking do it all on one line

**Pull request merged**